torch/_inductor/utils.py

```diff
@@ -429,35 +431,103 @@ def aggregate_origins(node_schedule):
         return set()
 
 
-def get_fused_kernel_name(node_schedule, descriptive_names):
+def get_module_name_from_meta(meta_dict):
+    mod_name = ""
+    torch_op = ""
+    if "nn_module_stack" in meta_dict:
+        # nn_module_stack is set in multiple places and can be organized
+        # differently.  The key is always the module name.  The value
+        # can either be a tuple or a string.  When it is a tuple, the
+        # last element of the tuple contains the torch_op.  When it is
+        # a string, the string is the torch_op.
+        # Example contents of nn_module_stack
+        # {'L__self___conv1': ("L['self'].conv1",
+        #                 <class 'torch.nn.modules.conv.Conv2d'>)}
+        # {'L__self___bn3': ("L['self'].bn3",
+        #                 <class 'torch.nn.modules.batchnorm.BatchNorm2d'>)}
+        # The key is the unique module name.  The value contains the
+        # torch_op type eg. <class 'torch.nn.modules.batchnorm.BatchNorm2d'>
+        mod_name, extra = list(meta_dict["nn_module_stack"].items())[-1]
+        if isinstance(extra, tuple):
+            torch_op = extra[-1]
+        else:
+            torch_op = extra
+
+        # Clean up the module name so it has similar format to
+        # the output of torch.nn.Module.named_modules()
+        # Examples for mod_name clean up.
+        # Before: 'getattr_L__self___layer4___1___relu'
+        # After: 'L.layer4.1.relu'
+        # Before: 'getattr_L__self___layer4___0___downsample_1'
+        # After: L.layer4.0.downsample_1
+        mod_name = re.sub(r"___", ".", mod_name)
+        mod_name = re.sub(r".*(L)__self", r"\g<1>", mod_name)
+
+        # Clean up the torch_op string by removing the
+        # <class > tag.
+        # Example <class 'torch.nn.modules.conv.Conv2d'> is reduced to
+        # torch.nn.modules.conv.Conv2d
+        res = re.search(r"<class\s+'(\S+)'>", str(torch_op))
+        if res:
+            torch_op = res.group(1)
+
+    return mod_name, torch_op
+
+
+def get_origin_op_info(node_schedule, descriptive_names):
```

Comment:
can you add a docstring to all new functions ?

Reply:
thanks for reviewing @eellison , working on the revisions.